staging/src/k8s.io/legacy-cloud-providers/azure/azure.go

```diff
@@ -851,6 +856,11 @@ func (az *Cloud) updateNodeCaches(prevNode, newNode *v1.Node) {
 		if _, hasExcludeBalancerLabel := newNode.ObjectMeta.Labels[v1.LabelNodeExcludeBalancers]; hasExcludeBalancerLabel {
 			az.excludeLoadBalancerNodes.Insert(newNode.ObjectMeta.Name)
 		}
+
+		// If not in ready state, add to excludeLoadBalancerNodes cache.
+		if !isNodeReady(newNode) {
+			az.excludeLoadBalancerNodes.Insert(newNode.ObjectMeta.Name)
+		}
```

Comment:
else
  az.excludeLoadBalancerNodes.Delete(newNode.ObjectMeta.Name)

Reply:
I am hesitating a bit. Say you have a *ready* node labeled with `v1.LabelNodeExcludeBalancer`: in the previous if-block, the node would be added to `az.excludeLoadBalancerNodes` (line 857), and with your else statement, it would be then removed from the list, right?

The condition we're looking for when adding a node to `az.excludeLoadBalancerNodes` is: 
- it has the managedByAzureLabel=false OR 
- it has the v1.LabelNodeExcludeBalancers label OR 
- it is not ready. 

With that else statement a node will be included in `az.excludeLoadBalancerNodes` only if it isn't ready.

Reply:
I forgot to say, a node that /becomes/ ready will be removed from `az.excludeLoadBalancerNodes` in line 825: `			az.excludeLoadBalancerNodes.Delete(prevNode.ObjectMeta.Name)`

Reply:
counting on seeing every previousNode state is risky proposition.  We build controllers based on level driven logic.  I agree with your comment about writing to excludeloadbalancernodes needing more conditions, but I think the newNode state should be the driving factor.